templates/cluster/vsphere-standalone-cp/templates/vspheremachinetemplate-controlplane.yaml

```
@@ -17,7 +17,10 @@ spec:
           networkName: {{ .Values.controlPlane.network }}
           {{- if .Values.ipamEnabled  }}
           addressesFromPools:
-            - {{- toYaml .Values.ipPool.data | nindent 14 }}
+            - {{- toYaml .Values.ipPool.config | nindent 14 }}
+          {{- end }}
+          {{- with .Values.nameservers }}
+          nameservers: {{ . }}
```

```suggestion
          nameservers: {{ toYaml . | nindent 12 }}
```

I've checked the rendering myself. Here is the working one